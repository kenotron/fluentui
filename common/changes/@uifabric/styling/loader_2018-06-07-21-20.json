{
  "changes": [
    {
      "comment": "",
      "packageName": "@uifabric/styling",
      "type": "none"
    }
  ],
  "packageName": "@uifabric/styling",
  "email": "ken@gizzar.com"
}