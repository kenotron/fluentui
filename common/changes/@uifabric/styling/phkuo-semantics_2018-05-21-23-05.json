{
  "changes": [
    {
      "packageName": "@uifabric/styling",
      "comment": "cleanup semantic slots",
      "type": "patch"
    }
  ],
  "packageName": "@uifabric/styling",
  "email": "phkuo@microsoft.com"
}