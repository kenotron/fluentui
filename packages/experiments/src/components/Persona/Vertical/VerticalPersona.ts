import { VerticalPersonaView } from './VerticalPersona.view';
import { VerticalPersonaStyles } from './VerticalPersona.styles';
import { IVerticalPersonaProps, IVerticalPersonaStyles } from './VerticalPersona.types';
import { createStatelessComponent } from '../../../Foundation';

export const VerticalPersona: React.StatelessComponent<IVerticalPersonaProps> = createStatelessComponent<
  IVerticalPersonaProps,
  IVerticalPersonaStyles
>({
  displayName: 'VerticalPersona',
  view: VerticalPersonaView,
  styles: VerticalPersonaStyles
});
