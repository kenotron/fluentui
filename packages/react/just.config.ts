import { task, series, nodeExecTask, webpackCliTask, preset, resolveCwd } from '@fluentui/scripts';

preset();

task(
  'mf',
  series(webpackCliTask({ webpackCliArgs: ['--config', 'webpack.mf.config.js', '--progress'] }), () => {
    const httpServer = resolveCwd('http-server/bin/http-server');
    return nodeExecTask({ args: [httpServer, '-p', '2345', './dist'] });
  }),
);
