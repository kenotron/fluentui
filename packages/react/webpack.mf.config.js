// @ts-check
const resources = require('../../scripts/webpack/webpack-resources');
const webpack = require('webpack');
const { webpackMerge } = require('just-scripts');

const BUNDLE_NAME = 'fluentui-react';

let config = resources.createConfig(BUNDLE_NAME, false, {
  resolve: {
    extensions: ['.ts', '.tsx', '.js'],
  },

  module: {
    rules: [
      {
        test: /\.css$/,
        include: /node_modules/,
        use: ['style-loader', 'css-loader'],
      },
      {
        test: [/\.tsx?$/],
        use: {
          loader: 'ts-loader',
          options: {
            experimentalWatchApi: true,
            transpileOnly: true,
          },
        },
        exclude: [/node_modules/, /\.scss.ts$/, /\.test.tsx?$/],
      },
      {
        test: /\.scss$/,
        enforce: 'pre',
        exclude: [/node_modules/],
        use: [
          {
            loader: '@microsoft/loader-load-themed-styles', // creates style nodes from JS strings
          },
          {
            loader: 'css-loader', // translates CSS into CommonJS
            options: {
              esModule: false,
              modules: true,
              importLoaders: 2,
            },
          },
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: ['autoprefixer'],
              },
            },
          },
          {
            loader: 'sass-loader',
          },
        ],
      },
    ],
  },

  // @ts-ignore
  plugins: [
    new webpack.container.ModuleFederationPlugin({
      name: 'fluentuiReact',
      filename: 'remoteEntry.js',
      exposes: {
        './Button': './src/Button',
      },
    }),
  ],
  output: {
    filename: `${BUNDLE_NAME}.mf.js`,
  },
})[0];

delete config.entry;

module.exports = config;
