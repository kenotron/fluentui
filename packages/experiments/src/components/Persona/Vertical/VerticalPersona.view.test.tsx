import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { IProcessedStyleSet } from '@uifabric/styling';

import { IVerticalPersonaStyles } from './VerticalPersona.types';
import { VerticalPersonaView } from './VerticalPersona.view';

// These tests will ensure that your styles regions have classname representation in snapshot output.
// (Also, classNames is a required prop for views, so we have to supply it for tests.)
const testVerticalPersonaClassNames: IProcessedStyleSet<IVerticalPersonaStyles> = {
  root: 'test-cn-root',
  primaryText: 'test-cn-text',
  secondaryText: 'test-cn-secondaryText',
  coin: 'test-cn-coin',
  subComponentStyles: {}
};

// Views are just pure functions with no statefulness, which means they can get full code coverage
// with snapshot tests exercising permutations of the props.
describe('VerticalPersonaView', () => {
  it('renders correctly with only a text', () => {
    const tree = renderer.create(<VerticalPersonaView vertical text="James Bond" classNames={testVerticalPersonaClassNames} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a text and secondary text', () => {
    const tree = renderer
      .create(
        <VerticalPersonaView vertical text="James Bond" secondaryText="Super secret agent" classNames={testVerticalPersonaClassNames} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the coin with the passed coinProps', () => {
    const tree = renderer
      .create(
        <VerticalPersonaView
          vertical
          text="James Bond"
          secondaryText="Super secret agent"
          classNames={testVerticalPersonaClassNames}
          coin={{ initials: 'MI6', coinColor: 'red' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
