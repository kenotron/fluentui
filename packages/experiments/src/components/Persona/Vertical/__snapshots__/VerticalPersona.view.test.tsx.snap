// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalPersonaView renders correctly with a text and secondary text 1`] = `
<div
  className="test-cn-root"
>
  <div
    className=

        {
          align-items: center;
          background-color: #5E4B8B;
          border-radius: 50%;
          color: white;
          display: flex;
          height: 48px;
          justify-content: center;
          position: relative;
          width: 48px;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: block;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 19.2px;
            font-weight: 400;
            moz-osx-font-smoothing: grayscale;
            overflow: hidden;
            text-overflow: ellipsis;
            webkit-font-smoothing: antialiased;
            white-space: nowrap;
          }
    >
      JB
    </span>
  </div>
  <span
    className=
        test-cn-text
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          webkit-font-smoothing: antialiased;
        }
  >
    James Bond
  </span>
  <span
    className=
        test-cn-secondaryText
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          webkit-font-smoothing: antialiased;
        }
  >
    Super secret agent
  </span>
</div>
`;

exports[`VerticalPersonaView renders correctly with only a text 1`] = `
<div
  className="test-cn-root"
>
  <div
    className=

        {
          align-items: center;
          background-color: #5E4B8B;
          border-radius: 50%;
          color: white;
          display: flex;
          height: 48px;
          justify-content: center;
          position: relative;
          width: 48px;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: block;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 19.2px;
            font-weight: 400;
            moz-osx-font-smoothing: grayscale;
            overflow: hidden;
            text-overflow: ellipsis;
            webkit-font-smoothing: antialiased;
            white-space: nowrap;
          }
    >
      JB
    </span>
  </div>
  <span
    className=
        test-cn-text
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          webkit-font-smoothing: antialiased;
        }
  >
    James Bond
  </span>
</div>
`;

exports[`VerticalPersonaView renders the coin with the passed coinProps 1`] = `
<div
  className="test-cn-root"
>
  <div
    className=

        {
          align-items: center;
          background-color: red;
          border-radius: 50%;
          color: white;
          display: flex;
          height: 48px;
          justify-content: center;
          position: relative;
          width: 48px;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: block;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 19.2px;
            font-weight: 400;
            moz-osx-font-smoothing: grayscale;
            overflow: hidden;
            text-overflow: ellipsis;
            webkit-font-smoothing: antialiased;
            white-space: nowrap;
          }
    >
      MI6
    </span>
  </div>
  <span
    className=
        test-cn-text
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          webkit-font-smoothing: antialiased;
        }
  >
    James Bond
  </span>
  <span
    className=
        test-cn-secondaryText
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          webkit-font-smoothing: antialiased;
        }
  >
    Super secret agent
  </span>
</div>
`;
